RedBoxMachine.java

```diff
@@ -12,48 +12,67 @@
 public class RedBoxMachine
 {
    //Create an instance variable to hold all of the DVDs.
-
+   ArrayList<DVD> library = new ArrayList<DVD>();
+   private int num;
    /** the list of DVDs */
-   
+
    /** Constructs a Redbox Machine and fills it with DVDs
     *  Reads the file MovieList.txt so make sure that the
     *  file is in the same folder as the RedboxMachine.class
     *  file.
     */
-    
+
    public RedBoxMachine()
    {
       // Complete the constructor.
       // Leave this method. It will fill the machine with DVDs.
       fillMachine();
    }
-   
+
    /** Searches for the movie with the provided title and returns
     *  the position of the DVD in the list if the DVD is found and
     *  -1 if the DVD is not in the list.
     *  @param title the title of the DVD being searched for.
     *  @return the index of the DVD in the list if present and
     *          -1 if the DVD is not in the list.
     */
+
    public int searchForMovie(String title)
    {
+      // for loop to check each index of the array list to search for movie title
+      for (int i = 0; i < library.size(); i++ )
```

Comment:
Use another variable to store the size so you aren't calling it every time through the loop.
`for (int i = 0, n = library.size(); i < n; i++)`